README.md

```diff
@@ -1,10 +1,105 @@
+Sesh is a session managment library written in Go. It uses a postgres table to track current sessions and their expiration, and it logs all session lifecycle events. It was created to fulfil the following requirements:
 
-## Dev Setup
+1. Sessions can be ended server-side, immediately rejecting all further requests from that session
+2. Only one session can be active at a time, if you login while you have a session active the old one will be ended.
+3. The browser stores the session in an HttpOnly cookie, minimzing the attack surface area for intercepting the session
+4. All session lifecycle events are logged: creation, destruction, reuse, and invalid requests.
 
-1. start a postgres server
-2. create a .env file with `cp .env.example .env`, set values to match your postgres config
-3. run `make reset_test_db`
-4. run `make test`
+## Configuration
+
+1. Run the migration included in ./migrations to set up the `sessions` table in your postgres db.
+2. Instantiate the sesh.Sessions struct
+
+```
+    dbConnection, err := sqlx.Open("postgres", dbConnectionURL)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to database using sqlx.Open: %w", err)
+	}
+
+    seshLogger := AuthLogger{}
+	sessions := sesh.NewSessions(dbConnection, seshLogger, 5*time.Minute, false)
+```
+
+There are several options, described below.
+
+3. Pass the Sessions struct to anywhere that needs it. Likely your router and your login/logout handlers.
+
+## Usage
+
+There are 5 places in your code where you need to interact with sesh once it's configured.
+
+1. Login
+2. Middleware for protected routes:
+3. Extracting the session id inside protected handlers
+4. Logout
+
+### Login
+
+When a user sucessfully authenticates, wether via username/password, oauth, client certs, etc. Create a new login and set the HttpOnly cookie like so:
+
+```
+    // With a valid accountID, we can begin a session.
+    _, err := sessions.UserDidAuthenticate(w, accountID.String())
+    if err != nil {
+        fmt.Println("Error Creating New Session", err)
+        http.Error(w, 500, http.StatusInternalServerError)
+        return
+    }
+
+}
+```
+
+This will create a new session associated with that AccountID and set the sesh cookie in the response writer. AccountID can be any string.
+
+### Middleware for protected routes
+
+To protect a route with sesh, add the sesh middleware to it.
+
+```
+    router := mux.NewRouter()
+    protectedRoutes := router.PathPrefix("/me").Subrouter()
+	protectedRoutes.Handle("", fetchCurrentUserHandler)
+    protectedRoutes.Handle("/dogs", fetchDogsHandler)
+
+	protectedRoutes.Use(sessions.AuthenticationMiddleware())
+```
+
+The middleware will grab the sesh cookie from the request, check that the session with that ID is valid, and add the Session struct to the context. If any part of that fails, it will log, write an error to the response, and not call any futher http handlers.
```

Comment:
```suggestion
The middleware will grab the sesh cookie from the request, check that the session with that ID is valid, and add the Session struct to the context. If any part of that fails, it will log, write an error to the response, and not call any further http handlers.
```